src/main/java/edu/cornell/kfs/concur/rest/authentication/ConcurAuthenticationFilter.java

```diff
@@ -0,0 +1,73 @@
+package edu.cornell.kfs.concur.rest.authentication;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.sys.context.SpringContext;
+
+import edu.cornell.kfs.concur.ConcurConstants;
+import edu.cornell.kfs.concur.service.ConcurAuthenticationService;
+
+public class ConcurAuthenticationFilter implements Filter {
+
+    @Override
+    public void destroy() {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+            ServletResponse response, FilterChain arg2) throws IOException,
+            ServletException {
+
+        if (servletRequest instanceof HttpServletRequest) {
+            HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
+
+            String authorizationHeader = httpServletRequest
+                    .getHeader(ConcurConstants.AUTHORIZATION_PROPERTY);
+
+            if (StringUtils.isBlank(authorizationHeader)) {
+                if (response instanceof HttpServletResponse) {
+                    HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+                    httpServletResponse
+                            .setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+                }
+                return;
```

Comment:
This return and the other doesn't seem quite right to me; I'm wondering if you could rework the code a bit to eliminate them? Perhaps the code after this if could be in an else?